java-squid/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -118,20 +115,8 @@ private static boolean isAnnotatedCheckForNull(MethodInvocationTree syntaxNode)
   }
 
   private static String getName(Tree syntaxNode) {
-    String name = "";
-    ExpressionTree expressionTree = null;
-    if (syntaxNode.is(Tree.Kind.MEMBER_SELECT)) {
-      expressionTree = ((MemberSelectExpressionTree) syntaxNode).expression();
-    } else if (syntaxNode.is(Tree.Kind.SWITCH_STATEMENT)) {
-      expressionTree = ((SwitchStatementTree) syntaxNode).expression();
-    }
-    if (expressionTree != null) {
-      if (expressionTree.is(Tree.Kind.IDENTIFIER)) {
-        name = ((IdentifierTree) expressionTree).name();
-      } else if (expressionTree.is(Tree.Kind.METHOD_INVOCATION)) {
-        name = ((MethodInvocationTree) expressionTree).symbol().name();
-      }
-    }
-    return name;
+    SyntaxTreeNameFinder finder = new SyntaxTreeNameFinder();
```

Comment:
@DidierBesset you could move this method as a static method in `SyntaxTreeNameFinder` so you will be able to use it with a single call.